### PR TITLE
Remove kapt from core-data

### PR DESCRIPTION
**Pull request by danimontoya**
- Remove kapt from core-data
- Remove higherkind annotations
- Add previously generated code to each typeclasses

### CODE REVIEW

File: arrow-core-data/src/main/kotlin/arrow/core/SortedMapK.kt [hunk 1]
```diff
@@ -1,12 +1,18 @@
 package arrow.core
 
 import arrow.Kind
-import arrow.higherkind
 import arrow.typeclasses.Applicative
 import arrow.typeclasses.Show
 import kotlin.collections.flatMap
 
-@higherkind
+class ForSortedMapK private constructor() { companion object }
```
Comment (raulraja) [inline arrow-core-data/src/main/kotlin/arrow/core/SortedMapK.kt:8]: All these should be deprecated without replacement now that they are spelled out in the sources.

**Review by raulraja — APPROVED** [review]
We need to deprecate those types but otherwise looks good, thanks @danimontoya !

File: arrow-core-data/src/main/kotlin/arrow/core/SortedMapK.kt (hunk 1 above)
Comment (danimontoya) [reply arrow-core-data/src/main/kotlin/arrow/core/SortedMapK.kt:8]: yes!! there are already ongoing PRs for Const, Eval, Ior deprecating them.. so they are being deprecated for each typeclass separately.. but in order to remove kapt and unblock one issue I have, this PR removing kapt is needed. Deprecation of all those will come soon too!

**Review by rachelcarmena — APPROVED** [review]
:raised_hands:

**Review by nomisRev — APPROVED** [review]

**Pull request merged**